include/zephyr/drivers/regulator.h

```diff
@@ -544,6 +555,52 @@ static inline int regulator_get_mode(const struct device *dev,
 	return api->get_mode(dev, mode);
 }
 
+/**
+ * @brief Set active discharge setting.
+ *
+ * @param dev Regulator device instance.
+ * @param active_discharge Active discharge enable or disable.
+ *
+ * @retval 0 If successful.
+ * @retval -ENOSYS If function is not implemented.
+ * @retval -errno In case of any other error.
+ */
+static inline int regulator_set_active_discharge(const struct device *dev,
+				     uint8_t active_discharge)
+{
+	const struct regulator_driver_api *api =
+		(const struct regulator_driver_api *)dev->api;
+
+	if (api->set_active_discharge == NULL) {
+		return -ENOSYS;
+	}
+
+	return api->set_active_discharge(dev, active_discharge);
+}
+
+/**
+ * @brief Get active discharge setting.
+ *
+ * @param dev Regulator device instance.
+ * @param[out] active_discharge Where active discharge will be stored.
+ *
+ * @retval 0 If successful.
+ * @retval -ENOSYS If function is not implemented.
+ * @retval -errno In case of any other error.
+ */
+static inline int regulator_get_active_discharge(const struct device *dev,
+				     uint8_t *active_discharge)
```

Comment:
```suggestion
				     bool *enabled)
```

note: Linux doesn't seem to offer a getter, is it really needed?

Reply:
correct, but just because Linux doesn't do it doesn't mean we also shouldn't do it. I can't really think of a reason not to have it.

Reply:
Linux is usually right, Zephyr... not so much ;-)